bdk_core/src/sparse_chain.rs

```diff
@@ -297,8 +297,7 @@ impl<I: ChainIndex> SparseChain<I> {
         self.prune_checkpoints();
     }
 
-    /// Clear the mempool list. Use with caution.
-    pub fn clear_mempool(&mut self) -> ChangeSet<I> {
+    pub fn clear_mempool_changeset(&self) -> ChangeSet<I> {
```

Comment:
Nit: for consistency.
```suggestion
    pub fn determine_clear_mempool_changeset(&self) -> ChangeSet<I> {
```